### DISCUSSION TITLE
How to improve the demo?

### DISCUSSION THREAD

**Discussion by gsvarovsky** [m-ld/m-ld-spec#53] [Discussion]
The website [demo](https://m-ld.org/demo/) doesn't get a lot of traffic. Some folks have said they don't know what it's demonstrating. Maybe it's too far departed from the core benefit of the technology...

How do we make it more engaging, and better demonstrate what m-ld is all about? Any ideas?

---

**Update:** the demo has been upgraded to the latest (v0.6) Javascript engine, and now transacts every character typed. We've also removed the bot, which was just annoying!

**Comment by mcalligator** [comment]
OK, first off, the demo is great for showcasing the concept and bring it to life.  I've a few ideas (both cosmetic and more fundamental) on how it could be improved:

1. After deleting a parent message, enable an orphaned child message to be attached to a new parent (cosmetic)
2. Enable creation of second and subsequent message threads in the same browser window without having to create an orphaned child
3. Provide a means for the message threads to be exported to a JPG/PNG, a text document, and a rich text document to allay concerns about data extraction.
4. Enable the messages to be persisted to say...a Solid Pod!

**Reply by gsvarovsky** [reply]
Thanks Angus. Great suggestions.

The first two actually comprise feedback on the (slightly over-precious) gestures available, because you can do these already but it's not obvious how!
- Dragging the blue plus button to another message creates a new connection
- Dragging the red cross button to another message deletes a connection that already exists between the messages

Export would be cool! and easy to do.

We could think of persistence to Solid as another kind of import/export. While a full Solid integration has been discussed (in #71), import/export is a lot simpler and could be really useful just by itself.
